lib/logstash/outputs/scalyr.rb

```diff
@@ -275,87 +276,90 @@ def multi_receive(events)
     # Just return and pretend we did something if running in noop mode
     return events if @noop_mode
 
-    start_time = Time.now.to_f
+    begin
+      start_time = Time.now.to_f
 
-    multi_event_request_array = build_multi_event_request_array(events)
-    # Loop over all array of multi-event requests, sending each multi-event to Scalyr
+      multi_event_request_array = build_multi_event_request_array(events)
+      # Loop over all array of multi-event requests, sending each multi-event to Scalyr
 
-    sleep_interval = @retry_initial_interval
-    batch_num = 1
-    total_batches = multi_event_request_array.length unless multi_event_request_array.nil?
+      sleep_interval = @retry_initial_interval
+      batch_num = 1
```

Comment:
Hm, I don't see this variable being increased anywhere? I assume it should be increased after processing the batch before moving to next one (if any), right?

Reply:
You're right, and this made me realize that the `sleep_interval` isn't reset to the initial value after each batch, its set to zero. Lots of weird small things like this.

Reply:
Wait `sleep_interval` is also not increased after a failure...